versions_test.go

```diff
@@ -322,22 +322,25 @@ func (s *VersionSuite) TestVersionConversionProducesExpectedVersionObjectsWithou
 	// should convert a string to a version object.
 	vString, err := ConvertVersion(expectedVersion)
 	s.NoError(err)
-	s.Equal(vString.source, expectedVersion)
+	s.Equal(vString.String(), expectedVersion)
 
 	// pass a pointer to a version object the converter
 	vVersionPointer, err := ConvertVersion(vString)
 	s.NoError(err)
-	s.Equal(vVersionPointer.source, expectedVersion)
+	s.Equal(vVersionPointer.String(), expectedVersion)
 
-	// pass a version object itself rather than a ref.
-	vVersionObj, err := ConvertVersion(*vVersionPointer)
+	// pass a legacy version object itself rather than a ref.
+	vVersionLegacy, ok := vVersionPointer.(*LegacyMongoDBVersion)
+	s.True(ok)
+	s.NotNil(vVersionLegacy)
```

Comment:
Can we `s.Require()` that these type-casting assertions are true so it doesn't panic the test later in case they're false? I find it oftentimes annoying to debug tests that assert something instead of require when a later assertion depends on a previous assumption (e.g. dereferencing a nil pointer without first requiring a non-nil pointer causes panic sometime further down in the test).

Reply:
Sure, i also am annoyed by these things